.github/workflows/unit-test-php.yml

```
@@ -0,0 +1,119 @@
+name: PHPunit with database
+
+on:
+  workflow_call:
+    inputs:
+      versions:
+        default: >-
+          ["8.1"]
+        description: "PHP versions"
+        required: false
+        type: string
+      php_extensions:
+        default: dom, curl, libxml, mbstring, mailparse, zip, pcntl, pdo, pdo_mysql
+        description: "PHP extensions to enable"
+        required: false
+        type: string
+      fail_on_low_coverage:
+        default: false
+        description: "Should this workflow fail when the coverage is below the minimum"
+        required: false
+        type: boolean
+      coverage_low:
+        default: 60
+        description: "Minimal required percentage of code that should be covered by tests, optionally fail when coverage is below minimum"
+        required: false
+        type: number
+      coverage_high:
+        default: 80
+        description: "Minimal optimal percentage of code that should be covered by tests"
+        required: false
+        type: number
+      pre_tests_script:
+        default:
+        description: "A bash script to run before running php unit"
+        required: false
+        type: string
+      env_file:
+        default: .env.testing
+        description: "Which env file to use as default"
+        required: false
+        type: string
+
+jobs:
+  php-unit-tests-with-database:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php: ${{ fromJson(inputs.versions) }}
+    env:
+      DB_CONNECTION: mysql
+      DB_USERNAME: root
+      DB_DATABASE: tests
+      DB_HOST: 127.0.0.1
+      DB_PORT: 3306
+      DB_PASSWORD:
+
+    services:
+      database:
+        image: mariadb:10
```

Use our own mariadb image
```suggestion
        image: docker.io/novoserve/mariadb10:2023.1.2
        credentials:
          username: ${{ secrets.dockerhub-username }}
          password: ${{ secrets.dockerhub-token }}
```